WordPress/gutenberg
# [Mobile] Pass the Gallery v2 Flag over from the editor

**Related PRs**
`gutenberg-mobile:` https://github.com/wordpress-mobile/gutenberg-mobile/pull/3713
`WordPress-iOS:` https://github.com/wordpress-mobile/WordPress-iOS/pull/16832
`WordPressKit-iOS:` https://github.com/wordpress-mobile/WordPressKit-iOS/pull/417





## Description

This passes the value of the editor setting `__unstableGalleryWithImageBlocks` from the Native Apps to the editor at initialization and after a settings update. `__unstableGalleryWithImageBlocks` is fetched via the `/wp-block-editor/v1/settings` API by the mobile apps alongside theme settings. The associated PRs attached here handle the iOS side of the changes.

📓 The Android side of this still needs to be handled.

## How has this been tested?



Testing this can be a bit involved: 

### Setup
1. A site running Gutenberg generated from this build. You can use `wp-env start` or generate a new bundle. 
2. The build from the associated iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/16832
3. On your site enable the Gallery Refactor by going to Gutenberg > Experminents > Check "Gallery Refactor"

### To Test
1. Open the editor. 
2. Add a Gallery Block
    - Expect the image to be an image block (i.e. the gallery is the refactored version). This will indicate that the Gallery is using the v2 version.
3. Navigate away from the editor
4. Take the device offline
5. Revalidate that the Gallery is using v2 version
6. Navigate away from the editor
7. Take the device online again
8. Toggle the flag back to false
9. Return to the editor and add a Gallery block again
    - Expect the image to not be in a nested image block. This will indicate we've reverted back to use the v1 setting. 

## Screenshots 

## Types of changes




New feature 

## Checklist:
